workflow/parser.py

```diff
@@ -0,0 +1,67 @@
+import os
+import subprocess
+
+
+class Parser:
+    def __init__(self, params):
+        self.params = params
+
+    def parse(self):
+
+        '''
+        - parse input config file 
+        - build TPS call 
+        - call TPS 
+        - gather outputs 
+        - return:
+            - list of outputs 
+            - output_folder
+            - out_label
+
+      
+        '''
+        # # grab TPS self.params 
+        build = []
+        OUT_FOLDER = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+        OUT_LABEL = ""
+
+        # add TPS run call as first arg
+        build.extend(["bash", "./scripts/run"])
+
+        # grab self.params from config
+        req = self.params[1]["TPS"][0]["required"]
+        op = self.params[1]["TPS"][1]["optional"]
+
+        # filter given args
+        filtered = {**req, **op}
+        args = {key:val for key, val in filtered.items() 
+                if filtered[key] != ""}
+
+                    
+        for key, val in args.items():
+            if key == "outlabel":
+                OUT_LABEL = str(val)
+            if key == "outfolder":
+                OUT_FOLDER = os.path.abspath(val)
+            if val == "1":
+                build.extend(['--' + key])
+            else:
+                k = "--" + key
+                build.extend([k, str(val)])
+
```

Comment:
I still prefer having these optional arguments that don't take values listed separate in a `- flags:` section of the config file.  Then we can loop over them separately and add them to `build` without arguments.